stage1/rootfs/diagexec/diagexec.c

```diff
@@ -180,19 +181,29 @@ static void load_env(const char *env)
 
 int main(int argc, char *argv[])
 {
-	const char *root, *cwd, *env, *exe;
-	exit_if(argc < 5,
-		"Usage: %s /path/to/root /work/directory /env/file /to/exec [args ...]", argv[0]);
+	const char *root, *cwd, *env, *uid_str, *gid_str, *exe;
+	char **args;
+	uid_t uid;
+	gid_t gid;
+	exit_if(argc < 7,
+		"Usage: %s /path/to/root /work/directory /env/file uid gid /to/exec [args ...]", argv[0]);
 
 	root = argv[1];
 	cwd = argv[2];
 	env = argv[3];
-	exe = argv[4];
+	uid_str = argv[4];
+	uid = atoi(uid_str);
+	gid_str = argv[5];
+	gid = atoi(gid_str);
+	args = &argv[6];
+	exe = args[0];
 
 	load_env(env);
 
-	pexit_if(chroot(root) == -1, "Chroot \"%s\" failed", root);
-	pexit_if(chdir(cwd) == -1, "Chdir \"%s\" failed", cwd);
+	pexit_if(*root && chroot(root) == -1, "Chroot \"%s\" failed", root);
+	pexit_if(*cwd && chdir(cwd) == -1, "Chdir \"%s\" failed", cwd);
```

Comment:
Is it expected that either cwd or root would be "" and should be silently ignored if so?

It seems incorrect for diagexec to have a **successful** code path which potentially doesn't attempt to chroot or chdir.